javatests/arcs/core/analysis/PredicateUtilsTest.kt

```diff
@@ -0,0 +1,134 @@
+package arcs.core.analysis
+
+import arcs.core.data.InformationFlowLabel
+import arcs.core.data.InformationFlowLabel.Predicate
+import com.google.common.truth.Truth.assertThat
+import java.util.BitSet
+import kotlin.test.assertFailsWith
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class PredicateUtilsTest {
+    enum class Labels {
+        A, B, C, D;
+
+        val asPredicate: Predicate.Label
+            get() = Predicate.Label(InformationFlowLabel.SemanticTag(name))
+    }
+    private val labels = enumValues<Labels>().map { it.name }
+    private val indicesMap: Map<InformationFlowLabel, Int> = enumValues<Labels>().map {
+        it.asPredicate.label to it.ordinal
+    }.toMap()
+
+    fun Predicate.asStringList(): List<String> {
+        return asSetOfBitSets(indicesMap).map { bitset ->
+            bitset.toString { bit ->
+                labels[bit]
+            }
+        }
+    }
+
+    @Test
+    fun prettyPrintBitSet() {
+        val setAC = BitSet(labels.size).apply {
+            set(Labels.A.ordinal)
+            set(Labels.C.ordinal)
+        }
+        assertThat(setAC.toString { bit -> labels[bit] }).isEqualTo("{A, C}")
+        assertThat(setAC.toString { _ -> "HAHA" }).isEqualTo("{HAHA, HAHA}")
+    }
+
+    @Test
+    fun LabelPredicate() {
```

Comment:
Nit: should start with lowercase (throughout file)

Reply:
oops! Fixed the case.